chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/SearchRequest.java

```diff
@@ -144,4 +147,62 @@ public Boolean getHasHadSurveillance() {
 	public void setHasHadSurveillance(Boolean hasHadSurveillance) {
 		this.hasHadSurveillance = hasHadSurveillance;
 	}
+	
+	@Override
+	public int hashCode() {

```

Comment:
I just spent a couple minutes searching but couldn't find the link where I found my specific algorithm. However, here is an article with some background information on the subject: http://crunchify.com/how-to-override-equals-and-hashcode-method-in-java/